com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/model/ModuleHook.java

```diff
@@ -0,0 +1,111 @@
+package com.sap.cloud.lm.sl.mta.model;
+
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+public class ModuleHook extends VersionedEntity implements VisitableElement, NamedElement {
+
+    public enum ModuleHookPhase {
+        APPLICATION_BEFORE_STOP, APPLICATION_BEFORE_START, NONE;
+
+        public static ModuleHookPhase fromString(String hookPhase) {
+            if (hookPhase.equals("application.before-start")) {
+                return ModuleHookPhase.APPLICATION_BEFORE_START;
+            }
+            if (hookPhase.equals("application.before-stop")) {
+                return ModuleHookPhase.APPLICATION_BEFORE_STOP;
+            }
+
+            throw new IllegalStateException(MessageFormat.format("Unsupported hook phase {0}", hookPhase));
+
+        }
+
+    }
+
+    public ModuleHook(int majorSchemaVersion) {
+        super(majorSchemaVersion);
+    }
+
+    public static ModuleHook copyOf(ModuleHook original) {
+        ModuleHook copy = new ModuleHook(original.majorSchemaVersion);
+        copy.name = original.name;
+        copy.type = original.type;
+        copy.phases = original.phases;
+        copy.parameters = new TreeMap<>(original.parameters);
+        copy.requiredDependencies = copyRequiredDependencies(original.requiredDependencies);
+        return copy;
+    }
+
+    private static List<RequiredDependency> copyRequiredDependencies(List<RequiredDependency> originals) {
+        return originals.stream()
+            .map(RequiredDependency::copyOf)
+            .collect(Collectors.toList());
+    }
+
+    private String name;
+    private List<ModuleHookPhase> phases;
+    private String type;
+    private Map<String, Object> parameters;
+    private List<RequiredDependency> requiredDependencies;
+
+    public static ModuleHook createV3() {
+        return new ModuleHook(3);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<ModuleHookPhase> getPhases() {
+        return phases;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public Map<String, Object> getParameters() {
+        return parameters;
+    }
+
+    public List<RequiredDependency> getRequiredDependencies() {
+        return requiredDependencies;
+    }
+
+    public ModuleHook setName(String name) {
+        this.name = name;
```

Comment:
Use the `this.<field>= ObjectUtils.defaultIfNull(<field>, this.<field>);` pattern in setters until I finish the MTA model refactoring and implement an actual builder pattern. :)
The `ObjectUtils.defaultIfNull` call protects users from accidentally overwriting the default values of fields with `null`s. The parsers, for example, can easily make that mistake.